packages/eslint-plugin/rules/__tests__/jsx-a11y-anchor-has-content.js

```diff
@@ -0,0 +1,332 @@
+/**
+ * External dependencies
+ */
+import { RuleTester } from 'eslint';
+
+/**
+ * Internal dependencies
+ */
+import rule from '../jsx-a11y-anchor-has-content';
+
+const ruleTester = new RuleTester( {
+	parserOptions: {
+		sourceType: 'module',
+		ecmaVersion: 6,
+		ecmaFeatures: {
+			jsx: true,
+		},
+	},
+} );
+
+ruleTester.run( '@wordpress/jsx-a11y-anchor-has-content', rule, {
+	valid: [
+		{
+			name: `Valid anchor with text content`,
+			code: `
+			( ) => {
+				( <div>
+					   { createInterpolateElement( __( 'This is a <a> link </a>.' ), {
+							   a: <a href="https://wordpress.org" />,
+					   } ) }
+				   </div>
+			   )
+		   };`,
+			options: [ { components: [ 'MyAnchor' ] } ],
+		},
+		{
+			name: `Valid anchor with text content and an extra attribute`,
+			code: `
+			( ) => {
+				( <div>
+					   { createInterpolateElement( __( 'help me <a> test with multiple words </a>.' ), {
+							   a: <a href="https://example.com" target="_blank" />,
+					   } ) }
+				   </div>
+			   )
+		   };`,
+		},
+		{
+			name: `Valid anchor with a custom component as content.`,
+			code: `
+			( ) => {
+				( <div>
+					   { createInterpolateElement( __( 'help me <a> <TextWrapper /> </a>.' ), {
+							   a: <a href="https://example.com" target="_blank" />,
+					   } ) }
+				   </div>
+			   )
+		   };`,
+		},
+		{
+			name: `Valid anchor without using createInterpolateElement`,
+			code: `
+			( ) => {
+				( <div>
+					   <a href="https://example.com"> test with multiple words </a>.
+				   </div>
+			   )
+		   };`,
+		},
+		{
+			name: `Invalid code, but out of scope for this rule (caught by jsx-a11y/anchor-has-content)`,
+			code: `
+			const a = ( ) => {
+				( <div>
+					{ createInterpolateElement( __( 'help me <a>Err</a>.' ), {
+						a: <a target="_blank"/>,
+				} ) }
+				   </div>
+			   )
+		   };`,
+		},
+		{
+			name: `Invalid usage of createInterpolateElement, ignored by the this rule`,
+			code: `
+			const a = ( ) => {
+				( <div>
+					{ createInterpolateElement( __( 'help me <a>Err</a>.' ), {} ) }
+				   </div>
+			   )
+		   };`,
+		},
+		{
+			name: `3 custom components through createInterpolateElement, all with valid anchors`,
+			code: `
+			import {
+				createInterpolateElement,
+			} from '@wordpress/element';
+			import { __ } from '@wordpress/i18n';
+			const withTranslate = ( ) => {
```

Comment:
```suggestion
			const withTranslate = () => {
```

Reply:
Also, should this start with a capital letter? Otherwise, React [will complain](https://iq.js.org/questions/react/why-should-component-names-start-with-capital-letter).

Same for below usages of `<withTranslate />`